pkg/envoy/proxy.go

```diff
@@ -134,12 +134,12 @@ func (e *envoy) args(fname string, bootstrapConfig string) []string {
 	if e.ProxyConfig.LogAsJSON {
 		startupArgs = append(startupArgs,
 			"--log-format",
-			`{"level":"%l","time":"%Y-%m-%dT%T.%fZ","scope":"envoy %n","msg":"%j"}`,
+			`{"level":"%l","time":"%Y-%m-%dT%T.%fZ","scope":"envoy %n","msg":"%j","loc":"%g:%#","tid":"%t"}`,
 		)
 	} else {
 		// format is like `2020-04-07T16:52:30.471425Z     info    envoy config   ...message..
 		// this matches Istio log format
-		startupArgs = append(startupArgs, "--log-format", "%Y-%m-%dT%T.%fZ\t%l\tenvoy %n\t%v")
+		startupArgs = append(startupArgs, "--log-format", "%Y-%m-%dT%T.%fZ\t%l\tenvoy %n:%t\t[%g:%#] %v")
```

Comment:
would it make sense to use `%@` instead?

Reply:
Not really, there're too many `config.cc` files in envoy.

Reply:
Is there many of the same filename AND emitting the same log on the same line? That feels unlikely. IMO the ideal is `package/filename` which is what the Go one uses, since then you have a lot more context without all the boilerplate, but I understand envoy doesn't support that (nor are their "packages" like Go)

Reply:
This is just the standard glog format that predates Go.

Reply:
The issue is not about what format it is or what is first, its consistency with the rest of Istio. On that note, I wanted to copy my comment from chat that we should only be emitting this if `--log_callers` enables it (for consistency). Otherwise we have some subset of logs that respect the flags and others that don't which leads to a poor UX.

one other note: this is not a unique envoy problem. We have solved bridging k8s, grpc, and Go stdlin logs using the same mechanisms and provided a consistent logging experience for Istio users rather than having 5 different logs with their own format and settings

Reply:
All of those are Go libraries, I think. Are we really going to claim the best log format for Java, etc? :) I'm fine implementing `log_callers` way if you can give me the link to where it is, and we can enable it by default.

Reply:
Actually upon further inspection I think its possible to do the log_callers thing but harder than expected, probably not reasonable to block on this